lib/rubocop/cop/style/to_json.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Avoids having both `JSON.generate` and `Object#to_json` in your codebase.
+      # They are equivalent, so for simplicity it makes sense to use only one.
+      # `to_json` is also what ActiveModel::Serializer uses, to it's easier to standardize on it.
+      #
+      # @example
+      #   # bad
+      #   JSON.generate(something)
+      #
+      #   # good
+      #   something.to_json
+      class ToJson < Base
+        extend AutoCorrector
+
+        MSG = 'Use `.to_json` instead of `JSON.generate`.'
+
+        # @!method json_generate?(node)
+        def_node_matcher :json_generate?, '(send (const nil? :JSON) :generate ...)'
+
+        def on_send(node)
+          return unless json_generate?(node)
+
+          add_offense(node) do |corrector|
+            _receiver, _method_name, *args = *node
+            data, opts = handle_args(args)
+            replacement = "#{data}.to_json"
+            replacement += "(#{opts})" if opts
+            corrector.replace(node, replacement)
+          end
+        end
+
+        private
+
+        def handle_args(args)
+          raise ArgumentError if args.size > 2
```

Comment:
I'm not sure we should raise in this scenario, as a host app could have something

```ruby
module MySerializer
  module JSON
    class << self
      def generate(foo, bar, biz)
```

and while false positives would be acceptable for that contrived case, I don't think raising an error if they attempt correction would be.

I think, instead, we'd want to detect non-standard arguments and not consider it an offense.

Reply:
Good point - we can just ignore cases where we have more than two arguments.